platform/platform_freebsd.cpp

```diff
@@ -0,0 +1,142 @@
+#include "platform/platform.hpp"
```

Comment:
The correct way is to create a separate platform_desktop_unix.{cpp|hpp} where all common Linux/FreeBSD code should reside. Linux is not FreeBSD and FreeBSD is not Linux. All Linux/FreeBSD-specific methods should be in platform_linux.cpp and platform_freebsd.cpp correspondingly.